sycl/doc/extensions/experimental/sycl_ext_oneapi_work_group_static.asciidoc

```diff
@@ -120,23 +113,21 @@ public:
   T* operator&() const noexcept;
 
 private:
-  T* ptr; // exposition only
-
+  T storage;
 };
 
 } // namespace sycl::ext::oneapi::experimental
 ----
 
-`T` must be one of the following:
-
-* A trivially constructible and trivially destructible type, or
-* An unbounded array of type `U`, where `U` is a trivially constructible and
-  trivially destructible type.
+`T` must be trivially constructible and trivially destructible.
```

Comment:
Should we also say here that `T` must be cv-unqualified?  If so, we should also add a static-assert to enforce this.

Reply:
I think we should, updated text, implementation and added test

Reply:
(Note: because we only have a trivial ctor, using a const type leads to an error anyway as the content cannot be initialized)